client/v3/snapshot/v3_snapshot.go

```diff
@@ -36,24 +36,25 @@ func hasChecksum(n int64) bool {
 	return (n % 512) == sha256.Size
 }
 
-// Save fetches snapshot from remote etcd server and saves data
-// to target path. If the context "ctx" is canceled or timed out,
-// snapshot save stream will error out (e.g. context.Canceled,
+// SaveWithVersion fetches snapshot from remote etcd server, saves data
+// to target path and returns server version. If the context "ctx" is cancelled or timed out,
+// snapshot save stream will error out (e.g. context.Cancelled,
```

Comment:
Great work. Nit, not sure why you changed to `context.Cancelled` or cancelled above. `context.Canceled` is the correct one. Thanks!